ui/tx/interpretation/TxInterpretation.tsx

```diff
@@ -89,23 +116,31 @@ const TxInterpretation = ({ summary, isLoading, className }: Props) => {
   const template = summary.summary_template;
   const variables = summary.summary_template_variables;
 
-  const variablesNames = extractVariables(template);
+  const variablesNamesInitial = extractVariables(template);
+
+  let intermediateResult = template;
+  variablesNamesInitial.forEach(name => {
+    if (variables[name] && variables[name].type === 'string') {
+      intermediateResult = intermediateResult.replace(`{${ name }}`, variables[name].value as string);
+    }
+  });
```

Comment:
Can we move this logic into a separate util function? Quite hard to figure out what is going on here